app/controllers/items_controller.rb

```diff
@@ -661,4 +749,23 @@ def enforce_versioning
     render status: :forbidden, text: 'Object cannot be modified in its current state.'
     false
   end
+
+  # ---
+  # Dor::Workflow utils
+
+  def dor_accession_error?(object, task)
```

Comment:
Some similar type stuff over here https://github.com/sul-dlss/argo/blob/develop/app/controllers/workflow_service_controller.rb#L42-L79 . Not sure if that can be combined, but just wanted to point it out.